### PR TITLE
Rework and add new Blame-Diff plots

### PR DESCRIPTION
**Pull request by boehmseb**
This PR mainly reworks the b_correlation_matrix plot and adds a new b_distribution_comparison plot.
It also adds a plot that applies different clustering algorithms on blame-diff data.

### CODE REVIEW

File: varats/plots/blame_diff_correlation_matrix.py [hunk 1]
```diff
@@ -60,37 +146,136 @@ def plot(self, view_mode: bool) -> None:
...
+
+        sns.set(style="ticks", color_codes=True)
+
+        df = pd.concat(dataframes)
+        df.set_index('revision', inplace=True)
+        df.drop(df[df.churn == 0].index, inplace=True)
+
+        _multivariate_grid(
+            x='churn',
+            y='num_interactions',
```
Comment (boehmseb) [inline varats/plots/blame_diff_correlation_matrix.py]: Any good ideas for making this configurable?
Should we use the extra-args mechanism of the plot-driver parser? Or generate multiple plots for all combinations?

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 20
           

Complexity increasing per file
==============================
- varats/plots/cluster_analysis.py  8
- varats/plots/blame_diff_correlation_matrix.py  3
         

Complexity decreasing per file
==============================
+ varats/data/databases/blame_diff_metrics_database.py  -2
         

Clones added
============
- varats/plots/cluster_analysis.py  1
- varats/plots/blame_diff_correlation_matrix.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5844099&bid=19134603)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- varats/plots/cluster_analysis.py  8
- varats/plots/blame_diff_correlation_matrix.py  3
         

Clones added
============
- varats/plots/cluster_analysis.py  1
- varats/plots/blame_diff_correlation_matrix.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5844099&bid=19134603)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/plots/cluster_analysis.py  8
- varats/plots/blame_diff_correlation_matrix.py  3
         

Clones added
============
- varats/plots/cluster_analysis.py  1
- varats/plots/blame_diff_correlation_matrix.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5844099&bid=19134603)

File: varats/plots/cluster_analysis.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+            plt.xticks(())
+            plt.yticks(())
+
+            plot_num += 1
+
+
+class BlameDiffClusterAnalysis(Plot):
+    """Performs different cluster algorithms on blame-diff data."""
+
+    NAME = "cluster_analysis"
```
Comment (vulder) [inline varats/plots/cluster_analysis.py]: could be a bit more specific to prevent name clashes

**Review by vulder — APPROVED** [review]
looks good overall ;)

File: varats/plots/blame_diff_correlation_matrix.py (hunk 1 above)
Comment (vulder) [reply varats/plots/blame_diff_correlation_matrix.py]: is this still relevant?

Comment (boehmseb) [reply varats/plots/blame_diff_correlation_matrix.py]: Yes.

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- varats/plots/blame_diff_correlation_matrix.py  3
- varats/plots/plot.py  1
- varats/plots/plots.py  7
- varats/plots/blame_diff_cluster_analysis.py  8
- varats/plots/commit_interactions.py  1
         

Complexity decreasing per file
==============================
+ varats/tools/driver_plot.py  -7
         

Clones added
============
- varats/plots/blame_diff_correlation_matrix.py  2
- varats/plots/paper_config_overview.py  1
- varats/plots/blame_diff_cluster_analysis.py  1
         

Clones removed
==============
+ varats/plots/blame_lorenz_curve.py  -2
+ varats/plots/repository_churn.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5844099&bid=19134603)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/plots/blame_diff_correlation_matrix.py  3
- varats/plots/plot.py  1
- varats/plots/commit_interactions.py  1
- varats/plots/plots.py  1
- varats/plots/blame_diff_cluster_analysis.py  8
         

Clones added
============
- varats/plots/blame_diff_correlation_matrix.py  2
- varats/plots/paper_config_overview.py  1
- varats/plots/blame_diff_cluster_analysis.py  1
         

Clones removed
==============
+ varats/plots/blame_lorenz_curve.py  -2
+ varats/plots/repository_churn.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5844099&bid=19134603)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/plots/blame_diff_cluster_analysis.py  8
- varats/plots/plot.py  1
- varats/plots/blame_diff_correlation_matrix.py  3
- varats/plots/plots.py  1
- varats/plots/commit_interactions.py  1
         

Clones added
============
- varats/plots/blame_diff_cluster_analysis.py  1
- varats/plots/blame_diff_correlation_matrix.py  2
- varats/plots/paper_config_overview.py  1
         

Clones removed
==============
+ varats/plots/repository_churn.py  -1
+ varats/plots/blame_lorenz_curve.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5844099&bid=19134603)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/plots/blame_diff_cluster_analysis.py  8
- varats/plots/blame_diff_correlation_matrix.py  3
- varats/plots/plot.py  1
- varats/plots/plots.py  1
- varats/plots/commit_interactions.py  1
         

Clones added
============
- varats/plots/blame_diff_cluster_analysis.py  1
- varats/plots/paper_config_overview.py  1
- varats/plots/blame_diff_correlation_matrix.py  2
         

Clones removed
==============
+ varats/plots/blame_lorenz_curve.py  -2
+ varats/plots/repository_churn.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5844099&bid=19134603)

**Pull request merged**